app/action.tsx

```diff
@@ -83,21 +81,17 @@ async function submitUserMessage(userInput: string): Promise<UIStateItem> {
             flightNumber: z.string().describe("the number of the flight"),
           })
           .required(),
-        render: async function* ({ flightNumber }) {
-          yield <Spinner />;
-
-          const flightInfo = await getFlightInfo(flightNumber);
-
+        render: function ({ flightNumber }) {
           aiState.done([
             ...aiState.get(),
             {
               role: "function",
               name: "get_flight_info",
-              content: JSON.stringify(flightInfo),
+              content: "TODO",
```

Comment:
I'm aware that this is not trivial to solve (one approach [here](https://github.com/unstubbable/mfng-ai-demo/pull/4)), so there's a chance that this may be an anti-pattern.

Reply:
But giving up colocation of data fetching in server components, when using them in an AI app (as shown in https://sdk.vercel.ai/docs/concepts/ai-rsc#nested-ui-streaming), would also be unfortunate.